crates/snapbox/src/data.rs

```diff
@@ -20,6 +22,14 @@ impl Data {
         }
     }
 
+    /// Mark the data as json (no post-processing)
+    #[cfg(feature = "json")]
+    pub fn json(raw: impl Into<Vec<serde_json::Value>>) -> Self {
```

Comment:
Should we make it easy to  create the value?  If so, from text, from Rust structures, or both?

Reply:
Can you give a few more details? I am a little uncertain what you are trying to ask

Reply:
If I were to do `assert.eq(expected, actual)`, I currently need to do something like
```rust
let expected = serde_json::from_str("{
}").unwrap();
let expected = Data::json(actual);

assert.eq(expected, actual)
```

Should we instead offer either
```rust
let expected = Data::json("{
}").unwrap();

assert.eq(expected, actual)
```
- However, `serde_json::object!` makes it easy to make json

or
```rust
let expected = ... some rust data ...;
let expected = Data::json(expected).unwrap();

assert.eq(expected, actual)
```
- Downside: we can't do wildcards

So maybe this needed but we should document this in the commit / PR and have our examples / tests use `serde_json::object!`

Reply:
You can do wildcard with 
```rust 
let expected = ... some rust data ...;
let expected = Data::json(expected).unwrap();

assert.eq(expected, actual)
```
`cargo_test_support` talks about it [here](https://github.com/rust-lang/cargo/blob/92ff4792755fef391663e978cd1de870552df1ad/crates/cargo-test-support/src/compare.rs#L509) and has an implementation for it [here](https://github.com/rust-lang/cargo/blob/92ff4792755fef391663e978cd1de870552df1ad/crates/cargo-test-support/src/compare.rs#L534)

The json would look like 
```json
{
    "important": "important",
    "wildcard": "{...}",
    "more_important_things": "stuff"
}
```

Reply:
Yes, which we can accomplish with `serde_json::json!()`.  However, we can't reproduce that with rust data structures and `Deserialize`.